hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/ContextTransactionProcessor.java

```diff
@@ -185,6 +191,8 @@ private CallOutcome maybeChargeFeesAndReturnOutcome(
             gasCharging.chargeGasForAbortedTransaction(
                     senderId, hederaEvmContext, rootProxyWorldUpdater, hevmTransaction);
         }
+
+        chargeOnFailedEthTxn(hevmTransaction);
```

Comment:
it seems that `gasCharging.chargeGasForAbortedTransaction` has the potential to charge the relay intrinsic gas if the transaction is an `EthereumTransaction`.  Given that right now `chargeGas` is only true if the transaction is a `ContractCall`, should we consider removing that clause from the if statement?  

I kind of do not like the fact that we now have two places where we charge someone fees (although the amounts are different - intrinsic gas vs $0.0001 converted to hbars).  Can the logic of the two calls be consolidated so that there is only one method that charges fees to the caller or the relay?

Reply:
For the first question I think the if statement you're mentioning here is out of scope for this task but we can definitely revisit to make sure it is correctly charging for gas.

For the second question - we can add a follow-up PR to address this. I'm tilting towards refactoring it but we should make sure that it is super clear that we might be charging for gas AND hapi fees.